app/routes/settings.js

```diff
@@ -69,6 +70,15 @@ export default TravisRoute.extend({
     });
   },
 
+  beforeModel() {
+    const repo = this.modelFor('repo');
+    const isAdmin = repo.get('permissions.admin');
+    if (!isAdmin) {
+      this.transitionTo('repo.index');
```

Comment:
I'm curious, doesn't it require to provide a `repo` model to transition?

Reply:
Yes, I tried that it works too :) I think you're right and it works since model is resolved on `repo` route level.